README.md

```diff
@@ -64,19 +69,159 @@ const main = defineCommand({
       description: "Use friendly greeting",
     },
   },
-  run({ args }) {
+  run({ args }) { // Command can be async
     console.log(`${args.friendly ? "Hi" : "Greetings"} ${args.name}!`);
   },
 });
 
 runMain(main);
 ```
 
+### Sub Commands
+
+You can define sub commands and attach them to main command to create a nested command structure. This is recursive so you can attach sub commands to sub commands, etc.
+
+```js
+import { defineCommand, runMain } from "citty";
+
+// First, you define a new command
+const sub = defineCommand({
+  meta: {
+    name: "sub",
+    description: "Sub command",
+  },
+  args: { // Sub commands can have their own arguments like any other command
+    name: {
+      type: "positional",
+      description: "Your name",
+      required: true,
+    },
+  },
+  run({ args }) { // Command can be async
+    console.log(`Hello ${args.name}!`);
+  },
+});
+
+// Then, you define a main command and attach sub command to it
+const main = defineCommand({
+  meta: {
+    name: "hello",
+    version: "1.0.0",
+    description: "My Awesome CLI App",
+  },
+  commands: {
+    sub, // Attach sub command to main command
+  },
+});
+
+runMain(main);
+```
+
+### Hooks
+
+`citty` supports a `setup` and `cleanup`  functions that are called before and after command execution. This is useful for setting up and cleaning up resources.
+
+Only the `setup` and `cleanup` functions from the command called are executed. For example, if you run `hello sub`, only the `setup` and `cleanup` functions from `sub` command are executed and not the ones from `hello` command.
+
+```js
+import { defineCommand, runMain } from "citty";
+
+const main = defineCommand({
+  meta: {
+    name: "hello",
+    version: "1.0.0",
+    description: "My Awesome CLI App",
+  },
+  setup() { // Setup function is called before command execution or before any sub command execution
+    console.log("Setting up...");
+  },
+  cleanup() { // Cleanup function is called after command execution or after any sub command execution
+    console.log("Cleaning up...");
+  },
+  run() {
+    console.log("Hello World!");
+  },
+});
+
+runMain(main);
+```
+
+### Lazy Load Commands
+
+For large CLI apps, you may want to only load the command that is being executed.
+
+First, create a command in a file and export it.
+
+```js
+import { defineCommand } from "citty";
+
+export default defineCommand({
+  meta: {
+    name: "sub",
+    description: "Sub command",
+  },
+  run({ args }) {
+    console.log(`Hello ${args.name}!`);
+  },
+});
+```
+
+Then, create the main command and import the sub command.
+
+```js
+const main = defineCommand({
+  meta: {
+    name: "hello",
+    version: "1.0.0",
+    description: "My Awesome CLI App",
+  },
+  commands: {
+    sub: () => import("./sub.js").then((m) => m.default), // Lazy Import Sub Command
+  },
+});
+```
+
+Now, when you run `hello sub`, the sub command will be loaded and executed. This avoid to load all commands at once.
```

Comment:
Don't we need to run with file extension like `node hello.ts sub`? I have reviewed thinking we need to specify extension

Reply:
Yes, during development, we need to specify the extension!